simpleflow/execute.py

```diff
@@ -181,7 +183,21 @@ def execute(*args, **kwargs):
                     close_fds=close_fds,
                     pass_fds=pass_fds,
                 )
-                rc = process.wait()
+
+                # Wait for process
+                if timeout:
+                    t_start = time.time()
```

Comment:
From: https://docs.python.org/2/library/signal.html
```
only the main thread can set a new signal handler, and the main thread will be the only one to receive signals
```

So unfortunately no, as we want to be able to execute an external Python program even out of the main thread.